auth/src/main/java/com/firebase/ui/auth/ui/email/RecoverPasswordHandler.java

```diff
@@ -0,0 +1,39 @@
+package com.firebase.ui.auth.ui.email;
+
+import android.app.Application;
+import android.arch.lifecycle.MutableLiveData;
+import android.support.annotation.NonNull;
+import android.support.annotation.RestrictTo;
+
+import com.firebase.ui.auth.util.data.AuthViewModelBase;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+public class RecoverPasswordHandler extends AuthViewModelBase {
+    private final MutableLiveData<RecoverPasswordProgressState> mProgressLiveData =
+            new MutableLiveData<>();
+
+    public RecoverPasswordHandler(Application application) {
+        super(application);
+    }
+
+    public MutableLiveData<RecoverPasswordProgressState> getProgressLiveData() {
```

Comment:
This method should return `LiveData` not `MutableLiveData` so that consumers don't mutate it.